frontend/src/DataFetchingSection.tsx

```diff
@@ -29,16 +32,24 @@ const DataFetchSection: React.FC = () => {
     timestamp: Math.floor(Date.now() / 1000),
   });
 
-  const handleFetchData = async () => {
+  const hexToJson = (hex: string) => {
+    const str = Buffer.from(hex.slice(2), 'hex').toString('utf8');
+    return JSON.parse(str);
+  };
+
+  // First function to handle contract write
+  const handleWriteContract = async () => {
     try {
       const mockData = generateMockData();
-      
+
       if (chain?.id === 31337) { // Anvil chain ID
+        setEndpoint('http://localhost:8080/graphql');
```

Comment:
I think the graphql endpoint should be set at page mount not when writing to the contract.

Since outputs are not tied to JSON-RPC requests, we can isolate the output-fetching process so that they don't have to make a write contract request before fetching a notice.